src/crimson/net/SocketMessenger.h

```diff
@@ -68,6 +89,19 @@ class SocketMessenger final : public Messenger {
   void unaccept_conn(SocketConnectionRef);
   void register_conn(SocketConnectionRef);
   void unregister_conn(SocketConnectionRef);
+
+  // required by sharded<>
+  seastar::future<> stop() {
+    return seastar::make_ready_future<>();
```

Comment:
probably it should `return shutdown()`?

Reply:
The usual practice to stop a sharded data-structure is to register it to `engine().at_exit()` and put it with its `start()` method, like https://github.com/scylladb/scylla/blob/master/main.cc#L419-L422 

But the problem is that we can't guarantee the order to stop them. For example, we might need to first stop messengers, then stop their dispatchers. So here the `shutdown()` serves this purpose, this means we don't put it into `stop()`.

Reply:
> But the problem is that we can't guarantee the order to stop them

we can. please see `reactor::run_exit_tasks()`, we can consider the exit tasks as a stack, when the seastar app exists, the tasks are popped out of the stack in a LIFO way. it is intuitive at ease to use.

Reply:
Another purpose here is to allow the msgr to be at the stand-by mode by calling [`wait()`](https://github.com/ceph/ceph/blob/5161060b5f471b87c84130647c14d416625f4b41/src/crimson/net/SocketMessenger.h#L108-L111), until `shutdown()` is called, the purpose is to implement the code like [ceph_osd.cc](https://github.com/ceph/ceph/blob/5161060b5f471b87c84130647c14d416625f4b41/src/ceph_osd.cc#L681-L687)

Also @majianpeng think it should be useful when to implement the server-side messenger 
 daemon during his performance tests.